src/sentry/api/endpoints/project_app_store_connect_credentials.py

```diff
@@ -411,3 +415,128 @@ def get(self, request: Request, project: Project, credentials_id: str) -> Respon
             },
             status=200,
         )
+
+
+class AppStoreConnectStatusEndpoint(ProjectEndpoint):  # type: ignore
+    """Returns a summary of the project's App Store Connect configuration
+    and builds.
+
+    ``GET projects/{org_slug}/{proj_slug}/appstoreconnect/status``
+
+    Response:
+    ```json
+    [
+        {
+            "id": "abc123",
+            "credentials": { status: "valid" } | { status: "invalid", code: "app-connect-forbidden-error" },
+            "pendingDownloads": 123,
+            "latestBuildVersion: "9.8.7" | null,
+            "latestBuildNumber": "987000" | null,
+            "lastCheckedBuilds": "YYYY-MM-DDTHH:MM:SS.SSSSSSZ" | null
+        },
+        {
+            "id": ...,
+            "credentials": ...,
+            "pendingDownloads": ...,
+            "latestBuildVersion: ...,
+            "latestBuildNumber": ...,
+            "lastCheckedBuilds": ...
+        },
+        ...
+    ]
+    ```
+
+    * ``pendingDownloads`` is the number of pending build dSYM downloads.
+
+    * ``latestBuildVersion`` and ``latestBuildNumber`` together form a unique identifier for
+      the latest build recognized by Sentry.
+
+    * ``lastCheckedBuilds`` is when sentry last checked for new builds, regardless
+      of whether there were any or no builds in App Store Connect at the time.
+    """
+
+    permission_classes = [ProjectPermission]
+
+    def get(self, request: Request, project: Project) -> Response:
+        config_ids = appconnect.AppStoreConnectConfig.all_config_ids(project)
+        statuses = []
+        for config_id in config_ids:
+            try:
+                symbol_source_cfg = appconnect.AppStoreConnectConfig.from_project_config(
+                    project, config_id
+                )
+            except KeyError:
+                continue
+
+            credentials = appstore_connect.AppConnectCredentials(
+                key_id=symbol_source_cfg.appconnectKey,
+                key=symbol_source_cfg.appconnectPrivateKey,
+                issuer_id=symbol_source_cfg.appconnectIssuer,
+            )
+
+            session = requests.Session()
+
+            try:
+                apps = appstore_connect.get_apps(session, credentials)
+            except appstore_connect.UnauthorizedError:
+                asc_credentials = {
+                    "status": "invalid",
+                    "code": AppConnectAuthenticationError().code,
```

Comment:
I plan to throw these errors directly in `/app` so the FE can use the same code to handle the validation messages returned by this endpoint as well as `/status`

Reply:
I'm not sure you need to instantiate the class here, can you not access the code attribute without instantiating?

Reply:
you're right about that, thanks for catching this!